From 22428420a0af3fb24f68f49b8ebe6b493450b600 Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Wed, 21 Jan 2015 18:54:12 +0800
Subject: [PATCH 238/255] atomisp2: IA_CSS_EVENT_TYPE_FW_ERROR is deprecated

It was replaced by IA_CSS_EVENT_TYPE_FW_ASSERT, so change it.

Change-Id: I2dd5d48e6039ab348398a10103e804b115f36937
Tracked-On: #H1503910281
Signed-off-by: Bin Han <bin.b.han@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/37232
Reviewed-by: GuoqingX Zhang <guoqingx.zhang@intel.com>
Reviewed-by: Tian Shu Qiu <tian.shu.qiu@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../pci/atomisp2/atomisp_driver/atomisp_compat_css20.c     | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index 86b5958..14bc008 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -4571,10 +4571,16 @@ int atomisp_css_isr_thread(struct atomisp_device *isp,
 	int i;
 
 	while (!atomisp_css_dequeue_event(&current_event)) {
-		if (current_event.event.type == IA_CSS_EVENT_TYPE_FW_ERROR) {
-			/* Received FW error signal, trigger WDT to recover */
-			dev_err(isp->dev, "%s: ISP reports FW_ERROR event, error code %d!!!!",
-				__func__, current_event.event.fw_error);
+		if (current_event.event.type ==
+			IA_CSS_EVENT_TYPE_FW_ASSERT) {
+			/*
+			 * Received FW assertion signal,
+			 * trigger WDT to recover
+			 */
+			dev_err(isp->dev, "%s: ISP reports FW_ASSERT event! fw_assert_module_id %d fw_assert_line_no %d\n",
+				__func__,
+				current_event.event.fw_assert_module_id,
+				current_event.event.fw_assert_line_no);
 			for (i = 0; i < isp->num_of_streams; i++)
 				atomisp_wdt_stop(&isp->asd[i], 0);
 			atomisp_wdt((unsigned long)isp);
-- 
1.9.1

